third_party/amd/python/triton_amd.cc

```diff
@@ -83,6 +83,8 @@ void init_triton_amd_passes_ttgpuir(py::module &&m) {
                      mlir::createTritonAMDGPUBlockPingpongPass);
   ADD_PASS_WRAPPER_3("add_stream_pipeline",
                      mlir::createTritonAMDGPUStreamPipelinePass, int, int, int);
+  ADD_PASS_WRAPPER_0("add_in_thread_transpose",
+                     mlir::createTritonAMDGPUInThreadTransposePass);
```

Comment:
This pass operates on funcOp. You need to `pm.addNestedPass<mlir::triton::FuncOp>`

Reply:
Good catch! This actually reminds me--we should add a simple test under `third_party/amd/python/` with the env var being on to check the generate ISA has `v_perm` and wide `ds_load`/`ds_write`?